hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/supervisor/workload/constraints/ZookeeperWorkloadConstraintsPathChildrenCache.java

```diff
@@ -0,0 +1,42 @@
+package pl.allegro.tech.hermes.consumers.supervisor.workload.constraints;
+
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.recipes.cache.ChildData;
+import org.apache.curator.framework.recipes.cache.PathChildrenCache;
+import org.apache.curator.framework.recipes.cache.PathChildrenCacheEvent;
+import org.apache.curator.framework.recipes.cache.PathChildrenCacheListener;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+class ZookeeperWorkloadConstraintsPathChildrenCache extends PathChildrenCache implements PathChildrenCacheListener {
+
+    private final Map<String, ChildData> cache = new ConcurrentHashMap<>();
+
+    ZookeeperWorkloadConstraintsPathChildrenCache(CuratorFramework curatorFramework, String basePath) {
+        super(curatorFramework, basePath, true);
+        getListenable().addListener(this);
+    }
+
+    Collection<ChildData> getChildrenData() {
+        return cache.values();
+    }
+
+    @Override
+    public void childEvent(CuratorFramework client, PathChildrenCacheEvent event) {
+        switch (event.getType()) {
+            case CHILD_ADDED:
+                cache.put(event.getData().getPath(), event.getData());
```

Comment:
IMHO these events should be used for preparing the domain classes we'll use in the future:

final Map<TopicName, Constraints> topicConstraints = new HashMap<>();
final Map<SubscriptionName, Constraints> subscriptionConstraints = new HashMap<>();

There is no need for computing it in each `getConsumersWorkloadConstraints()` call as we can store computed versions in our cache. What do you think?

Reply:
instead of `getChildrenData()` there would be simply `getConsumersWorkloadConstraints() -> new ConsumersWorkloadConstraints(topicsConstraints, subscriptionsConstraints)`

Reply:
You're right, that would be much better.

Reply:
Done